src/loopy/loopy.c

```diff
@@ -58,8 +69,8 @@ static int get_node(const char *image_path, char **loopdev) {
                 if (fd >= 0)
                         close(fd);
 
-                if (loop >= 0)
-                        close(loop);
+                if (loop >= 0 && success != 1)
```

Comment:
If this close() happens on failure you are going to close it again in single_mount